File: SHA1Pass/Sources/com/rkarp/sha1pass/components/CustomTextFieldSkin.java
package com.rkarp.sha1pass.components;

import com.rkarp.appcore.components.textfield.CustomTextField;
import javafx.scene.control.TextField;
import javafx.scene.control.skin.TextFieldSkin;

public class CustomTextFieldSkin extends TextFieldSkin {

    public CustomTextFieldSkin(CustomTextField passwordField) {
        super(passwordField);
    }

    @Override protected String maskText(String txt) {
        TextField textField = getSkinnable();

        int n = textField.getLength();
        StringBuilder passwordBuilder = new StringBuilder(n);
        for (int i=0; i<n; i++) {
            char BULLET = '\u2022';
            passwordBuilder.append(BULLET);
        }

        return passwordBuilder.toString();
    }
}

File: projects/JavaFXApps/core/src/main/java/net/rickiekarp/core/components/textfield/CustomTextFieldSkin.kt
package net.rickiekarp.core.components.textfield

import javafx.scene.control.skin.TextFieldSkin

class CustomTextFieldSkin(passwordField: CustomTextField) : TextFieldSkin(passwordField) {

    override fun maskText(txt: String): String {
        val textField = skinnable

        val n = textField.length
        val passwordBuilder = StringBuilder(n)
        for (i in 0 until n) {
            val bullet = '\u2022'
            passwordBuilder.append(bullet)
        }

        return passwordBuilder.toString()
    }
}